docs/madr/decisions/049-policy-on-namespace.md

```diff
@@ -0,0 +1,681 @@
+For convenience when referring to such type of policies we're going to call them **workload-owner policies**.
+The examples of workload-owner policies are MeshTrace, MeshMetric, MeshProxyPatch, MeshTrafficPermission, etc. 
+
+Top-level targetRef in Kuma policies always references a group of sidecars that are subjected to the change.
+New namespace-scoped policies are no exception, but we're going to implicitly add `k8s.kuma.io/namespace` and `kuma.io/zone` for **consumer** and **workload-owner policies**.
+
+```yaml
+# a valid consumer policy
+metadata:
+  namespace: frontend-ns
+  labels:
+    kuma.io/zone: zone-1
+spec:
+  targetRef:
+    kind: Mesh # despite it targets 'Mesh' the policy will be applied to pods in the 'frontend-ns' of 'zone-1'
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+        namespace: backend-ns
+---
+# workload-owner policy with 'from'
+metadata:
+  namespace: frontend-ns
+  labels:
+    kuma.io/zone: zone-1
+spec:
+  targetRef:
+    kind: Mesh # despite it targets 'Mesh' the policy will be applied to pods in the 'frontend-ns' of 'zone-1'
+  from:
+    - targetRef:
+        kind: Mesh
+```
+
+### Validation
+
+Policies with `to` array that have items for local and non-local namespaces shouldn't be allowed:
+
+```yaml
+# invalid policy example
+metadata:
+  namespace: finance-ns
+  labels:
+    kuma.io/zone: zone-1
+spec:
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+        namespace: finance-ns # this indicates the policy is a producer policy
+    - targetRef:
+        kind: MeshService
+        name: redis
+        namespace: redis-ns # this indicates the policy is a consumer policy
+```
+
+Policies that support both `from` and `to` and specify them at the same time shouldn't be allowed as well.
+
+### Syntactic sugar
+
+There are few things in policies syntax Kuma can figure out without requiring user to specify them explicitly.
+The syntactic sugar is very important as it allows user to not specify zone's name and namespace explicitly in the policy's yaml. 
+
+We don't want to change `spec` on the fly, because it might be surprising for user to see completely different resource in the store.
+Also, in case of the future changes in the logic, it's preferable to store the less explicit version. 
+But we probably have to implement a new feature in the Inspect API that allows viewing the most explicit version of the resource.
+
+#### Empty namespace in `to[].targetRef`
+
+If the `spec.to[idx].targetRef.namespace` is not specified then it's equal to `metadata.namespace`.
+
+#### Empty top-level targetRef
+
+* if the top-level targetRef is not specified in **producer policy**
+  then it's assumed to be `Mesh`
+* if the top-level targetRef is not specified in **consumer policy** or **workload-owner policy**
+  then it's assumed to be `MeshSubset` with `kuma.io/zone` and `k8s.kuma.io/namespace` tags.
```

Comment:
Also if you have partial MeshSubset .e.g:

```
targetRef:
  kind: MeshSubset
  tags:
    foo: bar
```

It gets rewritten as:


```
targetRef:
  kind: MeshSubset
  tags:
    foo: bar
    kuma.io/zone: foo
    k8s.kuma.io/namesapce: bar
```

Reply:
Yeah, it's right next to the sentence I sent:

> Top-level targetRef in Kuma policies always references a group of sidecars that are subjected to the change. New namespace-scoped policies are no exception, but we're going to implicitly add k8s.kuma.io/namespace and kuma.io/zone for consumer and workload-owner policies.
> 
> ```yaml
> # a valid consumer policy
> metadata:
>   namespace: frontend-ns
>   labels:
>     kuma.io/zone: zone-1
> spec:
>   targetRef:
>     kind: Mesh # despite it targets 'Mesh' the policy will be applied to pods in the 'frontend-ns' of 'zone-1'
>   to:
>     - targetRef:
>         kind: MeshService
>         name: backend
>         namespace: backend-ns
> ---
> # workload-owner policy with 'from'
> metadata:
>   namespace: frontend-ns
>   labels:
>     kuma.io/zone: zone-1
> spec:
>   targetRef:
>     kind: Mesh # despite it targets 'Mesh' the policy will be applied to pods in the 'frontend-ns' of 'zone-1'
>   from:
>     - targetRef:
>         kind: Mesh
> ```